pkg/templating/template_description_test.go

```diff
@@ -138,3 +138,148 @@ func Test_TemplateStringEnriched(t *testing.T) {
+		wantErr               bool
+		expectedEnrichedIssue *enrichedIssue
+	}{
+		{
+			name: "zero startTime returns err",
+			args: args{
+				option: EnrichedIssueWithScanStartTime(time.Time{}),
+			},
+			wantErr: true,
+		},
+		{
+			name: "non zero startTime returns no err",
+			args: args{
+				option: EnrichedIssueWithScanStartTime(acceptableTime),
+			},
+			wantErr: false,
+			expectedEnrichedIssue: &enrichedIssue{
+				ScanStartTime: acceptableTime.Format(time.RFC3339),
+			},
+		},
+		{
+			name: "zero firstFound returns err",
+			args: args{
+				option: EnrichedIssueWithFirstFound(time.Time{}),
+			},
+			wantErr: true,
+		},
+		{
+			name: "non zero firstFound returns no err",
+			args: args{
+				option: EnrichedIssueWithFirstFound(acceptableTime),
+			},
+			wantErr: false,
+			expectedEnrichedIssue: &enrichedIssue{
+				FirstFound: acceptableTime.Format(time.RFC3339),
+			},
+		},
+		{
+			name: "empty tool name returns err",
+			args: args{
+				option: EnrichedIssueWithToolName(""),
+			},
+			wantErr: true,
+		},
+		{
+			name: "valid tool name returns no err",
+			args: args{
+				option: EnrichedIssueWithToolName("some-tool"),
+			},
+			wantErr: false,
+			expectedEnrichedIssue: &enrichedIssue{
+				ToolName: "some-tool",
+			},
+		},
+		{
+			name: "empty confidence text returns err",
+			args: args{
+				option: EnrichedIssueWithConfidenceText(""),
+			},
+			wantErr: true,
+		},
+		{
+			name: "valid confidence text returns no err",
+			args: args{
+				option: EnrichedIssueWithConfidenceText("conf-text-1"),
+			},
+			wantErr: false,
+			expectedEnrichedIssue: &enrichedIssue{
+				ConfidenceText: "conf-text-1",
+			},
+		},
+		{
+			name: "empty severity text returns err",
+			args: args{
+				option: EnrichedIssueWithSeverityText(""),
+			},
+			wantErr: true,
+		},
+		{
+			name: "valid severity text returns no err",
+			args: args{
+				option: EnrichedIssueWithSeverityText("severity-text-1"),
+			},
+			wantErr: false,
+			expectedEnrichedIssue: &enrichedIssue{
+				SeverityText: "severity-text-1",
+			},
+		},
+		{
+			name: "0 count returns err",
+			args: args{
+				option: EnrichedIssueWithCount(0),
+			},
+			wantErr: true,
+		},
+		{
+			name: "positive count text returns no err",
+			args: args{
+				option: EnrichedIssueWithCount(420),
+			},
+			wantErr: false,
+			expectedEnrichedIssue: &enrichedIssue{
+				Count: 420,
+			},
+		},
+		{
+			name: "empty scan ID returns err",
+			args: args{
+				option: EnrichedIssueWithScanID(""),
+			},
+			wantErr: true,
+		},
+		{
+			name: "valid scan id returns no err",
+			args: args{
+				option: EnrichedIssueWithScanID("scan-1"),
+			},
+			wantErr: false,
+			expectedEnrichedIssue: &enrichedIssue{
+				ScanID: "scan-1",
+			},
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			var ei enrichedIssue
+			if err := tt.args.option(&ei); tt.wantErr {
+				require.Error(t, err)
+				return
+			}
+			assert.Equal(t, tt.expectedEnrichedIssue, &ei)
```

Comment:
Since we are not doing any other meaningful checks later, assert is fine I think

In this way, we wouldn't stop the test if one test case fails. In terms of devex I think it's a better approach as you can imagine a scenario where you have multiple failing tests and you can discover them on the first go test run rather than calling it as many times as the number of failing tests :)

Reply:
yes it doesn't make a different because the tests are using t.Run, however it keeps the code base consistent. we are using `require` as much as possible around to make sure that we don't have any tests failing in one place and then keep executing and producing more noise

Reply:
Don't feel strongly about this - refactored to use require